test/metabase/api/email_test.clj

```diff
@@ -26,24 +27,29 @@
   (testing "PUT /api/email"
     (tu/discard-setting-changes [email-smtp-host email-smtp-port email-smtp-security email-smtp-username
                                  email-smtp-password email-from-address]
-      ((user->client :crowberto) :put 200 "email" default-email-settings)
-      (is (= default-email-settings
-             (email-settings))))))
+      (with-redefs [email/test-smtp-connection (constantly {:error :SUCCESS})]
+        (is (= (assoc default-email-settings
+                      :with-corrections {})
+               (mt/user-http-request :crowberto :put 200 "email" default-email-settings)))
+        (is (= default-email-settings
+               (email-settings)))))))
```

Comment:
extra assertions, not needed, but I added them while debugging something else. Might as well leave them in :shrug: